src/test/java/com/puppycrawl/tools/checkstyle/xpath/XpathMapperTest.java

```diff
@@ -45,6 +48,65 @@ protected String getPackageLocation() {
         return "com/puppycrawl/tools/checkstyle/xpath/xpathmapper";
     }
 
+    @Test
+    public void testNodeOrdering() throws Exception {
+        final String xpath = "//METHOD_DEF[./IDENT[@text='callSomeMethod']]/SLIST/*";
+        final RootNode rootNode = getRootNode("InputXpathMapperAst.java");
+        final List<NodeInfo> nodes = getXpathItems(xpath, rootNode);
+        for (int i = 1; i < nodes.size(); i++) {
+            final NodeInfo curr = nodes.get(i);
+            final NodeInfo prev = nodes.get(i - 1);
+
+            if (curr.getLineNumber() == prev.getLineNumber()) {
+                assertWithMessage("Column number is not in document order")
+                    .that(curr.getColumnNumber())
+                    .isGreaterThan(prev.getColumnNumber());
+            }
+            else {
+                assertWithMessage("Line number is not in document order")
+                    .that(curr.getLineNumber())
+                    .isGreaterThan(prev.getLineNumber());
+            }
+        }
+    }
+
+    @Test
+    public void testParentChildOrdering() throws Exception {
```

Comment:
@gaurabdg 
`XpathMapperTest` should contain tests those test different xpath queries, like you did in `testNodeOrdering()`

Here you test `compareOrder` method of `ElementNode` class and there is no xpath query, so I would suggest to move this test to `ElementNodeTest`

Reply:
Okay, sure